File: structures-js/structures-api/src/api/idl/PageC3Type.ts
import {C3Type, ObjectC3Type} from '@kinotic/continuum-idl'

/**
 * This is a placeholder for the Page type.
 * This is not a complete type but a placeholder for the server to know what to expect.
 */
export class PageC3Type extends C3Type {

    /**
     * The {@link ObjectC3Type} of the content of the page
     * The actual page content will be a list of this type
     */
    public contentType: ObjectC3Type

    constructor(contentType: ObjectC3Type) {
        super('page')
        this.contentType = contentType
    }
}

File: structures-core/src/main/java/org/kinotic/structures/api/domain/idl/PageC3Type.java
package org.kinotic.structures.api.domain.idl;

import lombok.*;
import lombok.experimental.Accessors;
import org.kinotic.continuum.idl.api.schema.C3Type;
import org.kinotic.continuum.idl.api.schema.ObjectC3Type;

/**
 * This is a placeholder for the Page type.
 * This is not a complete type but a placeholder for the server to know what to expect.
 * Created by Navíd Mitchell 🤪 on 5/7/24.
 */
@Getter
@Setter
@Accessors(chain = true)
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
@ToString(callSuper = true)
public class PageC3Type extends C3Type {

    /**
     * The {@link ObjectC3Type} of the content of the page
     * The actual page content will be a list of this type
     */
    @ToString.Exclude
    private ObjectC3Type contentType = null;

}
